src/main/java/tools/map/xml/creator/MapXmlHelper.java

```diff
@@ -456,438 +461,328 @@ static void clearGameSettings() {
-            parsePropertyNode(property);
-          }
+        getTerritoryDefintionsMap().put(terrName, terrDef);
+      }
+
+      @Override
+      protected void handleConnection(Attributes attributes) {
+        String t1Name = attributes.getValue(0);
+        String t2Name = attributes.getValue(1);
+        if (t1Name.compareTo(t2Name) > 0) {
+          final String swapHelper = t1Name;
+          t1Name = t2Name;
+          t2Name = swapHelper;
         }
-        if (!getGameSettingsMap().isEmpty()) {
-          stepToGo = (getNotes().length() > 0 ? GameStep.MAP_FINISHED : GameStep.GAME_SETTINGS);
+        Set<String> t1Connections = getTerritoryConnectionsMap().get(t1Name);
+        if (t1Connections != null) {
+          t1Connections.add(t2Name);
+        } else {
+          t1Connections = Sets.newLinkedHashSet();
+          t1Connections.add(t2Name);
+          getTerritoryConnectionsMap().put(t1Name, t1Connections);
         }
       }
-    }
-    return stepToGo;
-  }
 
-  private static void initializeAll() {
-    MapXmlCreator.mapFolderLocation = null;
-    MapXmlCreator.mapImageFile = null;
-    MapXmlCreator.mapCentersFile = null;
-    mapXmlData.initialize();
-  }
+      @Override
+      protected void handleResource(Attributes attributes) {
+        getResourceList().add(attributes.getValue(0));
+      }
 
-  private static void putNodesToProductionFrontiers(final NodeList productionChildNodes) {
-    for (int i = 0; i < productionChildNodes.getLength(); ++i) {
-      final Node productionRule = productionChildNodes.item(i);
-      if (productionRule.getNodeName().equals(XML_NODE_NAME_PRODUCTION_RULE)) {
-        parseProductionRuleNode(productionRule.getChildNodes());
-      } else if (productionRule.getNodeName().equals(XML_NODE_NAME_PRODUCTION_FRONTIER)) {
-        final String playerName =
-            productionRule.getAttributes().getNamedItem(XML_ATTR_ATTACHMENT_NAME_NAME).getNodeValue().substring(10);
-        final ArrayList<String> frontierRules = new ArrayList<>();
-        final NodeList productionFrontierChildNodes = productionRule.getChildNodes();
-        for (int j = 0; j < productionFrontierChildNodes.getLength(); ++j) {
-          final Node productionFrontierChildNode = productionFrontierChildNodes.item(j);
-          if (productionFrontierChildNode.getNodeName().equals(XML_NODE_NAME_FRONTIER_RULES)) {
-            frontierRules
-                .add(productionFrontierChildNode.getAttributes().getNamedItem(XML_ATTR_ATTACHMENT_NAME_NAME)
-                    .getNodeValue().substring(3));
-          }
-        }
-        getProductionFrontiersMap().put(playerName, frontierRules);
+      @Override
+      protected void handlePlayer(Attributes attributes) {
+        final String playerNameAttr = attributes.getValue(XML_ATTR_PLAYER_NAME_NAME);
+        getPlayerNames().add(playerNameAttr);
+        getPlayerInitResourcesMap().put(playerNameAttr, 0);
+        // TODO: add logic for optional value
```

Comment:
Is this TODO still outstanding?  Are we ever really going to get to it, or when would we need it? I suspect we could remove this line and the commented out code below.